### PR TITLE
Add a Pulumi program for locally testing the online store application

### PR DESCRIPTION
**Pull request by scottslowe**
This PR adds a TypeScript Pulumi program (found in `develop/pulumi`) that allows Zephyr developers to stand up the online store application on their local laptops using the Docker provider.

### CODE REVIEW

**Comment by cnunciato** [comment]
When I run `pulumi up` in `develop/pulumi`, I get an error about a missing `srcRepoPath`:

```
Diagnostics:
  pulumi:pulumi:Stack (zephyr-local-dev):
    error: Missing required configuration variable 'zephyr-local:srcRepoPath'
        please set a value using the command `pulumi config set zephyr-local:srcRepoPath <value>`
```

What should this be? Should it fall back (or maybe always be set?) to something like `../src/ui`?

**Comment by scottslowe** [comment]
> When I run `pulumi up` in `develop/pulumi`, I get an error about a missing `srcRepoPath`:
> 
> ```
> Diagnostics:
>   pulumi:pulumi:Stack (zephyr-local-dev):
>     error: Missing required configuration variable 'zephyr-local:srcRepoPath'
>         please set a value using the command `pulumi config set zephyr-local:srcRepoPath <value>`
> ```
> 
> What should this be? Should it fall back (or maybe always be set?) to something like `../src/ui`?

Yes, try using `../src/ui`. That _should_ work. During my testing, the Docker provider was quite particular about paths and such. If that works, I can set it as a default and also update any associated documentation.

**Comment by scottslowe** [comment]
The correct setting should be `../..`, assuming you're running from the `pulumi/develop` folder. I also think this _should_ be a default setting, so I'll make that change.

File: develop/pulumi/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+This directory contains a Pulumi program for local testing (i.e., to build container images from source and deploy them to a local Docker daemon).
+
+Before you can use this Pulumi program, you will need:
+
+1. The `pulumi` CLI installed
+2. A local Docker daemon running (using Docker Desktop for Windows/Mac is fine)
+3. NodeJS installed (this was tested with Node 18)
+4. Version 4.1 or greater of the Pulumi Docker provider (should be installed automatically when you run `pulumi up`)
+
+Once the prerequisites are met, then set configuration values (see the section titled "Configuration Values") and run `pulumi up`. Within a few minutes, all the containers will be deployed locally. You can access the Zephyr online store by navigating to `http://localhost:8888`.
```
Comment (cnunciato) [inline develop/pulumi/README.md:12]:
```suggestion
Once the prerequisites are met, set configuration values (see the section titled "Configuration Values") and run `pulumi up`. Within a few minutes, all the containers will be deployed locally. You can access the Zephyr online store by navigating to `http://localhost:8888`.
```

File: develop/pulumi/index.ts [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+        },
+    ],
+    ports: [
+        {
+            internal: 8080,
+            external: 8888,
+        },
+    ],
+    restart: "always",
+}, { dependsOn: [ checkoutContainer, catalogContainer, ordersContainer, assetsContainer, cartsContainer ]});
```
Comment (cnunciato) [inline develop/pulumi/index.ts:450]: When this finished, my first question was _How do I see the running app?_ What do you think about exporting it to make that a little easier? 

```suggestion
}, { dependsOn: [ checkoutContainer, catalogContainer, ordersContainer, assetsContainer, cartsContainer ]});


// Export the frontend URL
export const frontendUrl = uiContainer.ports.apply(ports => `http://localhost:${ports![0].external}`);
```

**Review by cnunciato — APPROVED** [review]
Nice -- and worked on the first try with the latest code! 🎉 A few suggestions for you to consider, but nothing need block.

File: develop/pulumi/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/bin/
+/node_modules/
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply develop/pulumi/.gitignore:2]: Added `yarn.lock` in fdb6c7c

File: develop/pulumi/index.ts [hunk 4]
```diff
@@ -0,0 +1,450 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as docker from "@pulumi/docker";
+
+// Get configuration values
+const config = new pulumi.Config();
+const srcRepoPath = config.require("srcRepoPath");
+const mySqlPassword = config.require("mySqlPassword");
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply develop/pulumi/index.ts]: Added code for generating a random password in fdb6c7c

File: develop/pulumi/index.ts (hunk 2 above)
Comment (scottslowe) [reply develop/pulumi/index.ts:471]: Added this export in fdb6c7c

**Comment by scottslowe** [comment]
@cnunciato When you get a few minutes, can you pull down the latest changes and run it on both your Macs? I added some code for CPU awareness, and I want to be sure it works on both platforms.

**Pull request merged**